diff --git a/src/cargo-point.html b/src/cargo-point.html
index e30cea9..2b7f539 100644
--- a/src/cargo-point.html
+++ b/src/cargo-point.html
@@ -16,7 +16,7 @@
 		</div><!-- /.loading__content -->
 	</div><!-- /.loading-wrapper -->
 
-	<div class="wrapper">
+	<div class="wrapper wrapper--overflow">
 		{{> header }}
 
 		<section class="section-cargo">
@@ -247,8 +247,9 @@
 											<div class="js-select-custom select-custom section__select-custom">
 												<select id="example">
 													<optgroup label="United States - West Coast">
-														<option value="ca-los-angeles">California - Los Angeles</option>
+														<option disabled selected></option>
 
+														<option value="ca-los-angeles">California - Los Angeles</option>
 
 														<option value="ca-san-francisco">California - San Francisco</option>
 
@@ -298,6 +299,8 @@
 											<div class="js-select-custom select-custom section__select-custom">
 												<select id="example-3">
 													<optgroup label="United States - West Coast">
+														<option disabled selected></option>
+
 														<option value="ca-los-angeles">California - Los Angeles</option>
 
 														<option value="ca-san-francisco">California - San Francisco</option>
diff --git a/src/js/modules/select-custom.js b/src/js/modules/select-custom.js
index 8d309bc..8ba6eed 100644
--- a/src/js/modules/select-custom.js
+++ b/src/js/modules/select-custom.js
@@ -5,6 +5,19 @@
 /**
  * handleCustomSelect
  */
+// function handleCustomSelect($select) {
+// 	$select.each((idx, select) => {
+// 		const $customSelect = $(select);
+// 		const targetSelect = $customSelect.find('select');
+
+// 		targetSelect.select2({
+// 			placeholder: 'Select a location',
+// 			allowClear: true,
+// 			dropdownParent: $customSelect,
+// 		});
+// 	});
+// }
+
 function handleCustomSelect($select) {
 	$select.each((idx, select) => {
 		const $customSelect = $(select);
@@ -14,6 +27,49 @@ function handleCustomSelect($select) {
 			placeholder: 'Select a location',
 			allowClear: true,
 			dropdownParent: $customSelect,
+
+			templateResult: function (data) {
+				if (data.loading) return data.text;
+
+				if (data.element && data.element.parentElement && data.element.parentElement.label) {
+					data.parentText = data.element.parentElement.label;
+				}
+
+				return data.text;
+			},
+
+			matcher: function (params, data) {
+				if ($.trim(params.term) === '') return data;
+
+				if (!data.children) {
+					if (
+						data.text.toLowerCase().includes(params.term.toLowerCase()) ||
+						(data.parentText && data.parentText.toLowerCase().includes(params.term.toLowerCase()))
+					) {
+						return data;
+					}
+					return null;
+				}
+
+				const matchedChildren = [];
+
+				for (let i = 0; i < data.children.length; i++) {
+					const child = data.children[i];
+
+					if (
+						child.text.toLowerCase().includes(params.term.toLowerCase()) ||
+						(data.text && data.text.toLowerCase().includes(params.term.toLowerCase()))
+					) {
+						matchedChildren.push(child);
+					}
+				}
+
+				if (matchedChildren.length > 0) {
+					return $.extend({}, data, { children: matchedChildren });
+				}
+
+				return null;
+			},
 		});
 	});
 }
diff --git a/src/scss/components/_wrapper.scss b/src/scss/components/_wrapper.scss
index d7c95b8..2d4c35e 100644
--- a/src/scss/components/_wrapper.scss
+++ b/src/scss/components/_wrapper.scss
@@ -6,3 +6,8 @@
 	position: relative;
 	min-height: 100svh;
 }
+
+// wrapper overflow
+.wrapper--overflow {
+	overflow: hidden;
+}
diff --git a/src/scss/components/nav.scss b/src/scss/components/nav.scss
index 2ab28ab..2941437 100644
--- a/src/scss/components/nav.scss
+++ b/src/scss/components/nav.scss
@@ -147,6 +147,10 @@
 
 	> ul > li:nth-child(5) {
 		max-width: 15.4rem;
+
+		@include breakpoint-down(tablet) {
+			max-width: unset;
+		}
 	}
 
 	> ul > li:nth-child(2) > a:before {
diff --git a/src/scss/components/select-custom.scss b/src/scss/components/select-custom.scss
index 5ee5ce3..4e3a782 100644
--- a/src/scss/components/select-custom.scss
+++ b/src/scss/components/select-custom.scss
@@ -3,6 +3,9 @@
 \* ------------------------------------------------------------ */
 
 .select-custom {
+	@include breakpoint-down(mobile) {
+		font-size: 1.6rem;
+	}
 	.select2-container {
 		width: 100% !important;
 	}
@@ -16,6 +19,11 @@
 	.select2-container--open .select2-dropdown--below {
 		font-family: var(--ff-now-alt);
 	}
+
+	.select2-results__option .select2-results__option--disabled {
+		padding: 0;
+		margin: 0;
+	}
 }
 
 .select-custom .select2-container {
